### DISCUSSION TITLE
UI updates to pockethost.io

### DISCUSSION THREAD

**Discussion by benallfree** [pockethost/pockethost#1] [Q&A]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by brewhousedigital** [comment]
I'm a front-end developer, and would love to help out on the pockethost.io UI side if you're interested. I've built several sites in SvelteKit and just recently migrated one of our 75 page apps over to Pocketbase and I'm loving both platforms. Pockethost sounds like an awesome idea.

**Comment by benallfree** [comment]
Hi @brewhousedigital, your username looks familiar from somewhere. 

Anyway, yes I'd love for you to join in the fun. What ideas do you have?

**Reply by brewhousedigital** [reply]
I've been chatting a bit on the pocketbase discussions and I opened that issue for Pockethost yesterday. Maybe that's where!

I'm no help on the Go side but anything you need on the Sveltekit side, I can try to help with. Whether that's design, or new features. I'm more on the "design side" of the front-end field, rather than the comp sci part (like building an SDK for instance). Do you have a roadmap of features or a particular design you'd like Pockethost to resemble?

**Reply by benallfree** [reply]
That must be where I've seen you.

SvelteKit help would be awesome. Right now I'm using Sveltestrap, but I don't love it and I'm not a designer. Without even thinking about new features, what do you see about the current site UX/cosmetics that could be improved? And do you feel equipped to do some hands-on Svelte coding?

**Reply by brewhousedigital** [reply]
I cloned the repo and I'm running into ES module problems with `popper.js`. Looks like there's a couple of issues that might be related: [sveltekit](https://github.com/sveltejs/kit/issues/2161), [sveltestrap](https://github.com/bestguy/sveltestrap/pull/356). Are you experiencing this for the SvelteKit UI as well? I can only assume its related to Vite. Most ESM things are lol.

There are some other options for UI libraries that I can explore for you: https://madewithsvelte.com/ui-library if you aren't happy with Sveltestrap.

We could also just use the base version of Bootstrap - which would move a lot of the UI rendering to the CSS side instead of the JS side. For an example of that, check out this site: It uses SvelteKit and Bootstrap (along with Pocketbase!) https://battleborn.app/

The latest version of Bootstrap, 5.2, has switched to utilizing CSS variables which has made theming very easy. Let me know what you'd like me to try out.

![image](https://user-images.githubusercontent.com/66521220/195635509-ba39350a-acc2-4fec-903a-d2407784fe65.png)

**Reply by benallfree** [reply]
Yeah feel free to drop sveltestrap, that's fine! I did run into that popper.js issue, I can't remember what I did to fix it or why you're still hitting it since everything builds reliably for me now.

**Reply by benallfree** [reply]
https://github.com/bestguy/sveltestrap/issues/463

**Reply by benallfree** [reply]
I think it's an SSR issue

**Reply by benallfree** [reply]
Oh here's what I did https://github.com/sveltejs/kit/issues/2161#issuecomment-1250955253

> Add `"type": "module"` to `node_modules/@popperjs/core/package.json`

But the odd thing is that seems to not be needed because I just did a build on a fresh machine and didn't need to do that.

**Reply by benallfree** [reply]
But anyway yes please if you want to pick a different UI lib I would love that. Bootstrap feels kind of dated and we aren't that invested in it at this point.

**Comment by TechStudent10** [comment]
Hey! I'm a mainly React developer, but I've been looking into Svelte and there are some pages that I don't think look nice (couldn't thinking of a nicer way to put it). Mainly:

The dashboard:

![image](https://user-images.githubusercontent.com/76978184/209448528-5171123a-5da2-41af-9126-bd71213fc177.png)

And the instance:

![image](https://user-images.githubusercontent.com/76978184/209448542-28699d8c-6c0e-47db-a8b9-191b99faaf9f.png)

I may make some small mockups (I don't have Photoshop, so they would look like absolute crap), but I'd be willing to help.

**Reply by benallfree** [reply]
@TechStudent10 You're welcome to submit a frontend PR. Now would be a good time to do it, the frontend is pretty stable. Thank you for offering!
